goalie/mesh_seq.py

```diff
@@ -476,7 +486,9 @@ def get_checkpoints(self, solver_kwargs=None, run_final_subinterval=False):
         # Otherwise, solve each subsequent subinterval, in each case making use of the
         # previous checkpoint
         for i in range(N if run_final_subinterval else N - 1):
-            solutions = self.solver(i, checkpoints[i], **solver_kwargs)
+            self._reinitialise_fields(checkpoints[i])
+            self.solver(i, **solver_kwargs)
```

Comment:
I find this a bit odd. If calling `self.solver` this way initialises something then we should change the method name or introduce a new one that's more intuitive.

Reply:
Sorry, could you please clarify what you mean? :) Calling `self.solver` doesn't initialise the fields - calling `self._reinitialise_fields` (line above) does that. If I understood you right? :)

Reply:
My question is: what does the line `self.solver(...)` actually do? It's not at all obvious. I don't think we should be using properties in this way.

Reply:
Well, I assume that it was you who named it that way :) I did not change it (see the original line `solutions = self.solver(i, checkpoints[i], **solver_kwargs)`). I also didn't change its functionality, it still calls the solver defined in `get_solver`

Reply:
But I agree with you that it's not obvious and am happy to rename it. But I think I should do it in a separate PR since it has nothing to do with the changes I made here, as I tried to explain in the comment above

Reply:
Fair enough.

I think it just looks more confusing now because we are calling the method but we aren't using the return value. It would make more sense if it were called `solve`. But yeah, feel free to ignore for now.

Reply:
I see your point, yeah. I will see how #182 develops, where this confusion is already lesser, since the `get_checkpoints` method (which you are referring to here) calls `self._solve_forward` instead of `self.solver`.